### DISCUSSION TITLE
Aspect with dependencies and arguments

### DISCUSSION THREAD

**Discussion by jakublabno** [pamidur/aspect-injector#199] [Q&A]
Hi,
Is there any way to have Attribute and Aspect logic seperated into different classes?
The case is I want dependencies to be injected into aspect, but also I want my aspect Attribute accepts arguments.
By using factory it's not possible since arguments to attributes are being pass through constructor.

example

```
namespace Example
{
    public sealed class ResponseCode : System.Attribute
    {
        public readonly int Code;

        public ResponseCode(int code)
        {
            Code = code;
        }
    }
}
```
I want to read this attribute in my aspect definition class where I have also another dependencies. Factory method is not applicable to this.
I saw that I can use static factory, but I need my dependencies that were created in Startup class.
https://github.com/pamidur/aspect-injector/discussions/166

**Comment by pamidur** [comment]
It is exactly how it works. Merging attributes and aspects is just complimentary feature.
It is described here https://github.com/pamidur/aspect-injector/blob/master/docs/injection.md
Example is here https://github.com/pamidur/aspect-injector/blob/master/samples/src/Notify/NotifyAspect.cs

Basically you make your aspect like this:
```c#
[Aspect(Scope.Global, Factory = typeof(MyFactory))]
public class Aspect{
   ... here are your advices
}
```

and Attribute like this:
```c#
[Injection(typeof(Aspect))]
public class MyAttributeWithArgs{
 public string Text {get;set;}
}
```

and then follow this discussion on how to use it with DI 
https://github.com/pamidur/aspect-injector/discussions/166
